yarn-project/aztec-rpc/src/synchroniser/synchroniser.ts

```diff
@@ -20,33 +19,36 @@ export class Synchroniser {
     private log = createDebugLogger('aztec:aztec_rpc_synchroniser'),
   ) {}
 
-  public start(fromBlock = 1, take = 1, retryInterval = 1000) {
+  public start(from = 1, take = 1, retryInterval = 1000) {
     if (this.running) {
       return;
     }
 
     this.running = true;
-    let fromUnverifiedData = fromBlock;
 
     const run = async () => {
       while (this.running) {
         try {
-          // TODO: Blocks should be processed as part of getUnverifiedData
-          const blocks = await this.node.getBlocks(fromBlock, take);
-          await this.decodeBlocks(blocks);
-
-          const unverifiedData = await this.node.getUnverifiedData(fromUnverifiedData, take);
+          const unverifiedData = await this.node.getUnverifiedData(from, take);
           if (!unverifiedData.length) {
             await this.interruptableSleep.sleep(retryInterval);
             continue;
           }
 
+          // Note: If less than `take` unverified data is returned, then I fetch only that number of blocks.
+          const blocks = await this.node.getBlocks(from, unverifiedData.length);
+          if (blocks.length !== unverifiedData.length) {
```

Comment:
Presumably we could process Math.min(blocks.length, unverifiedData.length) of each, rather than consuming none at all?

Reply:
Addressed in fc07a576df57fc61fe8e795c4d1dad55f7b7e942.